src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -97,6 +102,17 @@ public void run() {
         logger.trace("PeriodicArchiver for {} running", rule.getRecordingName());
 
         try {
+            JsonArray response = getArchivedRecordings();
+            Iterator<Object> it = response.iterator();
+            
+            while (it.hasNext()) {
+                JsonObject entry = (JsonObject) it.next();
+                String fileName = entry.getString("name");
+                if (fileName.contains(this.rule.getRecordingName())) {
```

Comment:
I think we want a more robust check than `fileName.contains(this.rule.getRecordingName())` - this could lead to a lot of false positive matches if my rule has a really short/common/simple name, or if the rule name happens to equal or be a substring of a target's alias, etc.

Reply:
There is the `auto_` prefix here that seems to make this less likely, but I think it's still possible we see false positives.

Reply:
I noticed from the HTTP API docs that archived recordings are saved with the following filename format:

[$TARGET_NAME]_[$RECORDING_NAME]_[$DATE]T[$TIME]Z[.$COUNTER][.jfr]

In my opinion, the target name and recording name together can be used to accurately determine a recording as belonging to a specific rule, using a Pattern regex for example.

So now my question is what exactly is TARGET_NAME? Judging from the code used to fulfill a PATCH save request here: 

https://github.com/cryostatio/cryostat/blob/98924a38e879d1284dd4c57369951383f71abf32/src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingPatchSave.java#L139

it's not simply the `Rule.targetAlias` field so I would have to "calculate" it the same way. This seems redundant however, so maybe I should implement the map as described in the TODO in the link above? That way the map can be used in the `PeriodicArchiver` and when doing a PATCH save. Which class should I store this map in?

Reply:
That map would help avoid lookups/calculations but it would also need to be repopulated if the Cryostat instance goes down and comes back up later and has pre-existing archives contents, so I don't think it will end up helping you in the end, here.

The targetName part of the saved file is indeed somewhat ambiguous and can't always be mapped exactly back to a target uniquely. We could change this behaviour and always save the file using the service URI of a target instead, which would make them unique/unambiguous.

Another possibility would be to subdivide the archives. Currently all archived recording files are dumped into a flat directory. If we instead created directories within the archive corresponding uniquely to targets/ServiceRefs, then archived files could be placed into those directories and identified as unambiguously belonging to a target as well.

ex. right now we have things like:

```
archive/target1alias_recording1.jfr
archive/target2_recording1.jfr
```

This could be made into:

```
archive/$hashOfTarget1ServiceURI/target1alias_recording1.jfr
archive/$hashOfTarget2ServiceURI/target2_recording.jfr
```

This way we keep the human-friendly filenames but also have a machine-friendly way of grouping and identifying them.

Reply:
Sounds good, I'll take a look into modifying the archive directory structure.